src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -128,6 +139,19 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
                   write(6,*)' error retrieving pointer to ',trim(cvars3d(ic3)),' for en_bar'
                   call stop2(9993)
                end if
+
+               if (.not. l_use_dbz_directDA) then
+               ! cycle if additional variables are assigned in 'anavinfo' since
+               ! they are not assigned in the case of general_read_fv3_regional
+                 if (trim(cvars3d(ic3)) .eq. 'ql' .or. trim(cvars3d(ic3)) .eq. 'QL' ) cycle
+                 if (trim(cvars3d(ic3)) .eq. 'qi' .or. trim(cvars3d(ic3)) .eq. 'QI' ) cycle
+                 if (trim(cvars3d(ic3)) .eq. 'qr' .or. trim(cvars3d(ic3)) .eq. 'QR' ) cycle
+                 if (trim(cvars3d(ic3)) .eq. 'qs' .or. trim(cvars3d(ic3)) .eq. 'QS' ) cycle
+                 if (trim(cvars3d(ic3)) .eq. 'qg' .or. trim(cvars3d(ic3)) .eq. 'QG' ) cycle
+                 if (trim(cvars3d(ic3)) .eq. 'qnr' .or. trim(cvars3d(ic3)) .eq. 'QNR' ) cycle
+                 if (trim(cvars3d(ic3)) .eq. 'w' .or. trim(cvars3d(ic3)) .eq. 'W' ) cycle
+               end if
+
```

Comment:
This about a more general situation that may use those hydrometeors. Could we control those control variables through anavinfo.
We can check if "ql,qi,qr,qs,qg,qnr,w" are set in anavinfo, then run "general_read_fv3_regional_dirZDA" to read them in. Otherwise, don't read.

Reply:
I deleted these unnecessary codes according to your comment.